rest/rest-smartproxy/src/main/java/org/ow2/proactive_grid_cloud_portal/smartproxy/RestSmartProxyImpl.java

```diff
@@ -312,44 +303,66 @@ public void addEventListener(SchedulerEventListenerExtended listener, boolean my
     @Override
     public boolean uploadInputfiles(TaskFlowJob job, String localInputFolderPath)
             throws NotConnectedException, PermissionException {
+        try {
+            List<String> userSpaces = getUserSpaceURIs();
+            List<String> inputSpaces = Arrays.asList(Tools.dataSpaceConfigPropertyToUrls(job.getInputSpace()));
 
-        String userSpace = getLocalUserSpace();
-        String inputSpace = job.getInputSpace();
+            String remotePath = extractRemotePath(userSpaces, inputSpaces);
 
-        if (!inputSpace.startsWith(userSpace)) {
-            // NOTE: only works for USERSPACE urls
-            logger.warn("RestSmartProxy does not support data transfers outside USERSPACE.");
-            return false;
-        }
-        String remotePath = inputSpace.substring(userSpace.length() + (userSpace.endsWith("/") ? 0 : 1));
-        String jname = job.getName();
-        logger.debug("Pushing files for job " + jname + " from " + localInputFolderPath + " to " + remotePath);
-        TaskFlowJob tfj = job;
-        for (Task t : tfj.getTasks()) {
-            logger.debug("Pushing files for task " + t.getName());
-            List<String> includes = Lists.newArrayList();
-            List<String> excludes = Lists.newArrayList();
-
-            List<InputSelector> inputFilesList = t.getInputFilesList();
-
-            if (inputFilesList != null) {
-                for (InputSelector is : inputFilesList) {
-                    addfileSelection(is.getInputFiles(), includes, excludes);
-                }
+            if (remotePath == null) {
+                // NOTE: only works for USERSPACE urls
+                throw new IllegalArgumentException("Could not extract remote path using inputSpace=" + inputSpaces +
+                                                   " and userSpace=" + userSpaces);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal " and userSpace=" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)